src/headless-cms/components/CmsPageSearch/CmsPageSearch.tsx

```diff
@@ -0,0 +1,179 @@
+  page?: Page | undefined | null;
+}> = ({ page }): JSX.Element => {
+  if (page) {
+    return <CmsPageSearchFromSubPages page={page} />;
+  }
+  return <CmsPageSearchFromAllPages />;
+};
+
+/**
+ * Search a page from all the pages.
+ */
+const CmsPageSearchFromAllPages: React.FC = (): JSX.Element => {
+  const [searchResult, setSearchResult] = React.useState<Page[]>([]);
+  const [searchTerm, setSearchTerm] = React.useState('');
+  const cmsClient = useCMSClient();
+
+  // Search from all the pages
+  const { data: pageData, loading } = usePagesSearchQuery({
+    client: cmsClient,
+    variables: {
+      search: searchTerm ?? '',
+    },
+  });
+
+  // Update the search results
+  React.useEffect(() => {
+    if (!loading && pageData?.pages) {
+      const pages = pageData?.pages?.edges
+        ?.map((edge) => edge?.node as Page)
+        .filter((page) => !!page);
+      setSearchResult(pages ?? []);
+    }
+  }, [searchTerm, pageData, loading]);
+
+  return (
+    <div>
+      <CmsPageSearchForm setSearchTerm={setSearchTerm} />
+      <CmsPageSearchList pages={searchResult} />
+    </div>
+  );
+};
+
+/**
+ * Search a page from page's sub pages.
+ */
+const CmsPageSearchFromSubPages: React.FC<{
+  page: Page | undefined | null;
+}> = ({ page }): JSX.Element => {
+  const [searchResult, setSearchResult] = React.useState<Page[]>([]);
+  const [searchTerm, setSearchTerm] = React.useState('');
+  const cmsClient = useCMSClient();
+
+  // Search sub pages
+  const { data: pageData, loading } = useSubPagesSearchQuery({
+    client: cmsClient,
+    skip: !page?.uri,
+    variables: {
+      id: page?.uri ?? '',
+      idType: PageIdType.Uri,
+      search: searchTerm ?? '',
+    },
+  });
+
+  // Update the search results
+  React.useEffect(() => {
+    if (!loading && pageData?.page) {
+      const subPages = pageData?.page?.children?.edges
+        ?.map((edge) => edge?.node as Page)
+        .filter((page) => !!page);
+      setSearchResult(subPages ?? []);
+    }
+  }, [searchTerm, pageData, loading]);
+
+  return (
+    <div>
+      <CmsPageSearchForm page={page} setSearchTerm={setSearchTerm} />
+      <CmsPageSearchList pages={searchResult} />
+    </div>
+  );
+};
+
+/**
+ * Search form to search headless cms pages from all the pages or 1 page's sub pages.
+ */
+const CmsPageSearchForm: React.FC<{
+  page?: Page | undefined | null;
+  setSearchTerm: React.Dispatch<React.SetStateAction<string>>;
+}> = ({ page, setSearchTerm }): JSX.Element => {
+  const searchForm = (helperText: string) => (
+    <div className={styles.searchForm}>
+      <SearchInput
+        label="Search"
+        helperText={helperText}
+        searchButtonAriaLabel="Search"
+        clearButtonAriaLabel="Clear search field"
+        onSubmit={(submittedValue) => setSearchTerm(submittedValue)}
+      />
+    </div>
+  );
+
+  if (page) {
+    return searchForm(`Search from ${page.title} page's subpages...`);
+  }
+
+  return searchForm('Search from all the pages...');
```

Comment:
Abstracting to a function here seems little bit overkill here 🤔 
```
 <div className={styles.searchForm}>
      <SearchInput
        label="Search"
        helperText={page ? `Search from ${page.title} page's subpages...` : 'Search from all the pages...'}
        searchButtonAriaLabel="Search"
        clearButtonAriaLabel="Clear search field"
        onSubmit={(submittedValue) => setSearchTerm(submittedValue)}
      />
    </div>
```

Reply:
changed to 
```

  const helperText = page
    ? `Search from ${page.title} page's subpages...`
    : 'Search from all the pages...';

  return (
    <div className={styles.searchForm}>
      <SearchInput
        label="Search"
        helperText={helperText}
        searchButtonAriaLabel="Search"
        clearButtonAriaLabel="Clear search field"
        onSubmit={(submittedValue) => setSearchTerm(submittedValue)}
      />
    </div>
  );
```